ray-curator/ray_curator/stages/video/io/video_reader.py

```diff
@@ -266,15 +268,27 @@ def decompose(self) -> list[ProcessingStage]:
         Returns:
             List of processing stages: [FilePartitioningStage, VideoReaderStage]
         """
-        reader_stage = FilePartitioningStage(
-            file_paths=self.input_video_path,
-            files_per_partition=1,
-            file_extensions=[".mp4", ".mov", ".avi", ".mkv", ".webm"],
-            limit=self.video_limit,
+        if self.input_video_path.startswith("s3://"):
```

Comment:
Lets add a util like:
```
def is_remote_url(url: str) -> bool:
    fs, _ = url_to_fs(url)
    proto = fs.protocol[0] if isinstance(fs.protocol, (list, tuple)) else fs.protocol
    return proto not in (None, "file")
```

And then here: 

```suggestion

        if is_remote(self.input_video_path): 
```